### PR TITLE
i18n(zh-cn): Update `markdown-content.mdx` `rss.mdx` `server-side-rendering.mdx` `from-nextjs.mdx`

### PR DESCRIPTION
**Pull request by zhengbangbo**
#### What kind of changes does this PR include?

- Translated content

#### Description

Updated three files on the `guides` directory
- markdown-content.mdx
- rss.mdx
- server-side-rendering.mdx

Updated from-nextjs.mdx

### CODE REVIEW

**Comment by zhengbangbo** [comment]
@liruifengv pls review

File: src/content/docs/zh-cn/guides/markdown-content.mdx [hunk 1]
```diff
@@ -461,18 +463,65 @@ import { rehypeAccessibleEmojis } from 'rehype-accessible-emojis';
...
+    rehypePlugins: [rehypeAccessibleEmojis],
   },
 });
 ```
   
 请注意，默认情况下，`remarkToc`需要页面上的 “ToC” 或 “Table of Contents” [标题](https://github.com/remarkjs/remark-toc#optionsheading)（不区分大小写）才能显示目录。
 
+#### 标题 ID 和插件
+
+Astro 将 `id` 属性注入到 Markdown 和 MDX 文件中的所有标题元素（`<h1>` 到 `<h6>`），并提供 `getHeadings()` 实用程序，用于在 [Markdown 导出的属性](#exported-properties) 中检索这些 ID。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/markdown-content.mdx:476]:
```suggestion
Astro 将 `id` 属性注入到 Markdown 和 MDX 文件中的所有标题元素（`<h1>` 到 `<h6>`），并提供 `getHeadings()` 实用程序，用于在 [Markdown 导出的属性](#导出的属性) 中检索这些 ID。
```
There has a broken link.

File: src/content/docs/zh-cn/guides/markdown-content.mdx [hunk 2]
```diff
@@ -461,18 +463,65 @@ import { rehypeAccessibleEmojis } from 'rehype-accessible-emojis';
...
+    remarkPlugins: [ [remarkToc, { heading: "contents"} ] ],
+    rehypePlugins: [rehypeSlug, [rehypeAutolinkHeadings, { behavior: 'append' }]],
+  },
+}
+```
+
 ### 以编程方式修改 frontmatter
 
+:::note
+如果您使用的是 [内容集合](/zh-cn/guides/content-collections/)，请参阅[“使用 Remark 修改 Frontmatter”](/zh-cn/guides/content-collections/#modifying-frontmatter-with-remark)。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/markdown-content.mdx:519]:
```suggestion
如果你使用的是 [内容集合](/zh-cn/guides/content-collections/)，请参阅[“使用 Remark 修改 Frontmatter”](/zh-cn/guides/content-collections/#modifying-frontmatter-with-remark)。
```

**Review by liruifengv — CHANGES_REQUESTED** [review]

File: src/content/docs/zh-cn/guides/server-side-rendering.mdx [hunk 3]
```diff
@@ -227,4 +283,131 @@ if(!product) {
...
 
-服务器端点，也称为 **API 路由**，是从 `src/pages/` 文件夹中的 `.js`或 `.ts` 文件导出的特殊函数。该函数接收一个[端点上下文对象](/zh-cn/reference/api-reference/#端点上下文)并返回一个 [Response](https://developer.mozilla.org/zh-CN/docs/Web/API/Response)。 这是 SSR 的一个强大功能，API 路由能够在服务器上安全地执行代码。 要了解更多信息，请参阅我们的[端点指南](/zh-cn/core-concepts/endpoints/#服务器端点api-路由)。
+服务器端点，也称为 **API 路由**，是从 `src/pages/` 文件夹中的 `.js`或 `.ts` 文件导出的特殊函数。
+该函数接收一个[端点上下文对象](/zh-cn/reference/api-reference/#端点上下文)并返回一个 [Response](https://developer.mozilla.org/zh-CN/docs/Web/API/Response)。 这是 SSR 的一个强大功能，API 路由能够在服务器上安全地执行代码。 要了解更多信息，请参阅我们的[端点指南](/zh-cn/core-concepts/endpoints/#服务器端点api-路由)。
+
+### 流式处理
+
+浏览器本身支持 HTTP 流，其中文档被分解为块，按顺序通过网络发送，并按该顺序呈现在页面上。
+
+在此过程中，浏览器以增量方式使用 HTML：解析、呈现到 DOM 中和绘制。无论您是否有意流式传输 HTML，都会发生这种情况。网络状况可能会导致大型文档下载缓慢，等待数据获取可能会阻止页面呈现。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/server-side-rendering.mdx:293]:
```suggestion
在此过程中，浏览器以增量方式使用 HTML：解析、呈现到 DOM 中和绘制。无论你是否有意流式传输 HTML，都会发生这种情况。网络状况可能会导致大型文档下载缓慢，等待数据获取可能会阻止页面呈现。
```

File: src/content/docs/zh-cn/guides/server-side-rendering.mdx [hunk 4]
```diff
@@ -227,4 +283,131 @@ if(!product) {
...
+  <body>
+    <h2>一个名字</h2>
+    <p>{randomPerson.name.first}</p>
+    <h2>一个事实</h2>
+    <p>{factData.fact}</p>
+  </body>
+</html>
+```
+
+将 `await` 调用移动到较小的组件中，您可以利用 Astro 的流式传输。使用以下组件执行数据获取，Astro 可以先渲染一些 HTML，例如标题，然后在数据准备就绪时呈现段落。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/server-side-rendering.mdx:320]:
```suggestion
将 `await` 调用移动到较小的组件中，你可以利用 Astro 的流式传输。使用以下组件执行数据获取，Astro 可以先渲染一些 HTML，例如标题，然后在数据准备就绪时呈现段落。
```

File: src/content/docs/zh-cn/guides/server-side-rendering.mdx [hunk 5]
```diff
@@ -227,4 +283,131 @@ if(!product) {
...
+    <RandomName />
+    <h2>一个事实</h2>
+    <RandomFact />
+  </body>
+</html>
+```
+
+#### 直接包括 Promises
+
+您还可以直接在模板中包含 promises。它不会阻止整个组件，而是并行解析 promise，并且只阻止它后面的标记。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/server-side-rendering.mdx:361]:
```suggestion
你还可以直接在模板中包含 promises。它不会阻止整个组件，而是并行解析 promise，并且只阻止它后面的标记。
```

File: src/content/docs/zh-cn/guides/server-side-rendering.mdx [hunk 6]
```diff
@@ -227,4 +283,131 @@ if(!product) {
...
+  </body>
+</html>
+```
+
+在此示例中，`一个名称`将在加载 `personPromise` 和 `factPromise` 时渲染。
+一旦 `personPromise` 解决，`一个事实`就会出现，`factpromise` 将在加载完成后呈现。
+
+### 将 SSR 构建拆分为多个文件
+
+默认情况下，Astro 将所有页面构建到一个名为 `entry.mjs` 的单个文件中。您可以在构建选项中配置 [`split`](/zh-cn/reference/configuration-reference/#buildsplit)，改为为每个页面发出一个文件：
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/server-side-rendering.mdx:390]:
```suggestion
默认情况下，Astro 将所有页面构建到一个名为 `entry.mjs` 的单个文件中。你可以在构建选项中配置 [`split`](/zh-cn/reference/configuration-reference/#buildsplit)，改为为每个页面发出一个文件：
```

File: src/content/docs/zh-cn/guides/markdown-content.mdx [hunk 7]
```diff
@@ -1,21 +1,24 @@
...
+安装 [@astrojs/mdx 集成](/zh-cn/guides/integrations-guide/mdx/)后，Astro 还支持 [MDX](https://mdxjs.com/) (.mdx) 文件，这些文件带来了附加功能，例如支持 JavaScript 表达式和在 Markdown 中使用组件。
 
 使用一种或两种类型的文件来编写你的 Markdown 内容！
 
 ## Markdown 和 MDX 页面
 
 ### 内容集合
 
-你可以在 Astro 中的特殊的 `src/content/` 文件夹中管理您的 Markdown 和 MDX 文件。[内容集合](/zh-cn/guides/content-collections/)可帮助你组织内容、验证你的 frontmatter，并在处理你的内容时提供自动 TypeScript 类型安全。
+你可以在 Astro 中的特殊的 `src/content/` 文件夹中管理您的 Markdown 和 MDX 文件。[内容集合](/zh-cn/guides/content-collections/) 可帮助你组织内容、验证你的 frontmatter，并在处理你的内容时提供自动 TypeScript 类型安全。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/markdown-content.mdx:21]:
```suggestion
你可以在 Astro 中的特殊的 `src/content/` 文件夹中管理你的 Markdown 和 MDX 文件。[内容集合](/zh-cn/guides/content-collections/) 可帮助你组织内容、验证你的 frontmatter，并在处理你的内容时提供自动 TypeScript 类型安全。
```

File: src/content/docs/zh-cn/guides/markdown-content.mdx [hunk 8]
```diff
@@ -372,7 +374,7 @@ import {Content as PromoBanner} from '../components/promoBanner.md';
 
 #### 举例：动态页面路由
 
-相反，你可以使用[动态路由](/zh-cn/core-concepts/routing/#动态路由)动态生成页面。
+与其将 Markdown/MDX 文件放在 'src/pages/' 目录中来创建页面路由，不如[动态路由](/zh-cn/core-concepts/routing/#动态路由)。
 
 要访问 Markdown 内容，请将 `<Content/>` 组件传递给 Astro 页面的 `props`。 然后，您可以从 `Astro.props` 中检索该组件并在页面模板中渲染它。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/markdown-content.mdx:379]:
```suggestion
要访问 Markdown 内容，请将 `<Content/>` 组件传递给 Astro 页面的 `props`。 然后，你可以从 `Astro.props` 中检索该组件并在页面模板中渲染它。
```

File: src/content/docs/zh-cn/guides/markdown-content.mdx [hunk 9]
```diff
@@ -461,18 +463,65 @@ import { rehypeAccessibleEmojis } from 'rehype-accessible-emojis';
...
 });
 ```
   
 请注意，默认情况下，`remarkToc`需要页面上的 “ToC” 或 “Table of Contents” [标题](https://github.com/remarkjs/remark-toc#optionsheading)（不区分大小写）才能显示目录。
 
+#### 标题 ID 和插件
+
+Astro 将 `id` 属性注入到 Markdown 和 MDX 文件中的所有标题元素（`<h1>` 到 `<h6>`），并提供 `getHeadings()` 实用程序，用于在 [Markdown 导出的属性](#exported-properties) 中检索这些 ID。
+
+你可以通过添加注入 `id` 属性的 rehype 插件（例如 `rehype-slug`）来自定义这些标题ID。您的自定义 ID（而不是 Astro 的默认值）将反映在 HTML 输出和 `getHeadings()` 返回的项目中。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/markdown-content.mdx:478]:
```suggestion
你可以通过添加注入 `id` 属性的 rehype 插件（例如 `rehype-slug`）来自定义这些标题ID。你的自定义 ID（而不是 Astro 的默认值）将反映在 HTML 输出和 `getHeadings()` 返回的项目中。
```

File: src/content/docs/zh-cn/guides/markdown-content.mdx [hunk 10]
```diff
@@ -461,18 +463,65 @@ import { rehypeAccessibleEmojis } from 'rehype-accessible-emojis';
...
+
+默认情况下，Astro 会在您的 rehype 插件运行后注入 `id` 属性。如果你的自定义 rehype 插件之一需要访问 Astro 注入的 ID，你可以直接导入和使用 Astro 的 `rehypeHeadingIds` 插件。确保在任何依赖它的插件之前添加 `rehypeHeadingIds`：
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/markdown-content.mdx:480]:
```suggestion
默认情况下，Astro 会在你的 rehype 插件运行后注入 `id` 属性。如果你的自定义 rehype 插件之一需要访问 Astro 注入的 ID，你可以直接导入和使用 Astro 的 `rehypeHeadingIds` 插件。确保在任何依赖它的插件之前添加 `rehypeHeadingIds`：
```

File: src/content/docs/zh-cn/guides/markdown-content.mdx [hunk 11]
```diff
@@ -705,7 +699,7 @@ export default defineConfig({
 
 Astro 主要设计用于可以保存在项目目录中的本地 Markdown 文件。 但是，在某些情况下，您可能需要从远程源获取 Markdown。 例如，您可能需要在构建网站时（或在使用 [SSR](/zh-cn/guides/server-side-rendering/) 时，用户向网站发出请求时）从远程 API 获取并呈现 Markdown。
 
-**Astro 不包含对远程 Markdown 的内置支持！** 要获取远程 Markdown 并将其呈现为 HTML，您需要从 npm 安装并配置自己的 Markdown 解析器。 这将不会从您配置的 Astro 的内置 Markdown 和 MDX 设置中继承。 在将其添加到您的项目之前，请确保您了解这些限制。
+**Astro 不包含对远程 Markdown 的内置支持！** 要获取远程 Markdown 并将其呈现为 HTML，您需要从 npm 安装并配置自己的 Markdown 解析器。 这将**不会**从您配置的 Astro 的内置 Markdown 和 MDX 设置中继承。 在将其添加到您的项目之前，请确保您了解这些限制。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/markdown-content.mdx:702]:
```suggestion
**Astro 不包含对远程 Markdown 的内置支持！** 要获取远程 Markdown 并将其呈现为 HTML，你需要从 npm 安装并配置自己的 Markdown 解析器。 这将**不会**从你配置的 Astro 的内置 Markdown 和 MDX 设置中继承。 在将其添加到你的项目之前，请确保你了解这些限制。
```

File: src/content/docs/zh-cn/guides/server-side-rendering.mdx [hunk 12]
```diff
@@ -1,18 +1,31 @@
...
+
+- 在应用中实现登录状态的会话。
+- 从使用 `fetch` 动态调用的 API 呈现数据。
+- 使用 *适配器* 将站点部署到主机。
+
+如果需要，请考虑在 Astro 项目中启用服务器端渲染：
+
+- **API 端点**：SSR 使你能够创建特定页面，这些页面充当数据库访问、身份验证和授权等任务的 API 端点，同时对客户端隐藏敏感数据。
+- **受保护的页面**：如果需要根据用户权限限制对页面的访问，可以启用 SSR 来处理服务器上的用户访问。
+- **频繁更改的内容**：启用 SSR 可让您生成单个页面，而无需静态重建站点。当页面内容频繁更新时，这很有用。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/server-side-rendering.mdx:20]:
```suggestion
- **频繁更改的内容**：启用 SSR 可让你生成单个页面，而无需静态重建站点。当页面内容频繁更新时，这很有用。
```

File: src/content/docs/zh-cn/guides/server-side-rendering.mdx [hunk 13]
```diff
@@ -23,6 +36,33 @@ export default defineConfig({
 });
 ```
 
+然后，您可以在任何页面或路由中使用 export 语句选择退出默认渲染行为：
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/server-side-rendering.mdx:39]:
```suggestion
然后，你可以在任何页面或路由中使用 export 语句选择退出默认渲染行为：
```

File: src/content/docs/zh-cn/guides/server-side-rendering.mdx [hunk 14]
```diff
@@ -98,21 +138,21 @@ export default defineConfig({
 
 Astro 在默认情况下仍是静态网站生成器。但是一旦你启用了服务器端渲染适配器，**你的页面目录中的每条路由都会成为服务器渲染的路由**，一些新的功能就可以为你所用。
 
-### 混合渲染
-
-<Since v="2.0.0" />
+### 配置单个路由
 
-要启用 **混合渲染**，任何支持导出变量的[页面](/zh-cn/core-concepts/astro-pages/) 或[服务器端点](/zh-cn/core-concepts/endpoints/#服务器端点api-路由)（.astro、.mdx、.ts 或 .js）都可以选择加入预渲染。这些文件将在构建时被静态渲染，类似于默认的 static [`output`](/zh-cn/reference/configuration-reference/#output) 模式。
+`server` 和 `hybrid` 模式都允许服务器渲染的页面和端点，并将根据其默认模式渲染。但是，这两种模式都允许您将单个页面标记为 opt-out 此默认行为。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/server-side-rendering.mdx:143]:
```suggestion
`server` 和 `hybrid` 模式都允许服务器渲染的页面和端点，并将根据其默认模式渲染。但是，这两种模式都允许你将单个页面标记为 opt-out 此默认行为。
```

**Review by liruifengv — CHANGES_REQUESTED** [review]
The `from-nextjs.mdx` page has a broken link due to this change. Can you update the `from-nextjs.mdx` page?

**Review by liruifengv — COMMENTED** [review]
Some commit for en was missed. You can find them in https://github.com/withastro/docs/commits/main/src/content/docs/en/guides/server-side-rendering.mdx?since=2023-04-04T13:31:19.000Z

Such as:
- https://github.com/withastro/docs/pull/3395/files
- https://github.com/withastro/docs/commit/961716b926d7f477049fc5d858a054c53a45556e

**Comment by zhengbangbo** [comment]
Many linked documents in the `from-nextjs.mdx` file are not translated into Chinese

Such as:

- [.astro 文件的语法类似于 JSX (EN)](https://docs.astro.build/zh-cn/core-concepts/astro-syntax/#differences-between-astro-and-jsx)
- [条件渲染内容 (EN)](https://docs.astro.build/zh-cn/core-concepts/astro-syntax/#dynamic-html)
- [内容集合 (EN)](https://docs.astro.build/zh-cn/guides/content-collections/)
- [动态生成这些页面 (EN)](https://docs.astro.build/zh-cn/guides/content-collections/#generating-routes-from-content)
- [使用 Collections API 查询 (EN)](https://docs.astro.build/zh-cn/guides/content-collections/#querying-collections)
- [组件属性的完整列表 (EN)](https://docs.astro.build/zh-cn/guides/integrations-guide/image/#usage)

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx [hunk 15]
```diff
@@ -413,12 +413,18 @@ const randomUser = data.results[0];
...
 
 查看更多关于 [在 Astro 中设计样式](/zh-cn/guides/styling/) 的信息。
 
 ### 转换 Next 的 图像插件到 Astro
-
+  
+:::note
+Astro v3.0 将包括一个新的 `astro:assets` 模块，并将弃用现有的 `@astrojs/image` 集成。
+  
+访问 `astro:assets` 目前可以在实验标志下使用，并且是开始新 Astro 项目的推荐方式。如果你要迁移到 Astro v3.0 之前，我们建议你 [启用实验标志以使用 assets](/zh-cn/guides/assets/) 作为图像解决方案。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx:425]:
```suggestion
访问 `astro:assets` 目前可以在实验标志下使用，并且是开始新 Astro 项目的推荐方式。如果你要迁移到 Astro pre-v3.0，我们建议你 [启用实验标志以使用 assets](/zh-cn/guides/assets/) 作为图像解决方案。
```

**Review by liruifengv — APPROVED** [review]
A little change. LGTM

**Review by Yan-Thomas — APPROVED** [review]
LGTM! Great work everyone 🚀

**Pull request merged**